src/db/collections/application_state.rs

```diff
@@ -54,4 +58,25 @@ impl ApplicationStateCollection {
         .await?;
         Ok(())
     }
+
+    /// Gets the singleton application state.
+    pub async fn get_last_migration(&self) -> Result<Option<String>, Error> {
+        Ok(self
+            .find_one::<ApplicationStateDocument>(doc! {}, None)
+            .await?
+            .and_then(|doc| doc.last_migration))
+    }
+
+    /// Set the current version in the singleton application state.
+    pub async fn set_last_migration(&self, last_migration: &str) -> Result<(), Error> {
+        self.update_one(
+            doc! {},
+            doc! {
+                "$set": { "last_migration": last_migration }
```

Comment:
I would write the current `PKG_VERSION` that was used to perform the migration into the struct too.

Reply:
But no use-case

Reply:
It may help with debugging

Reply:
Feel free to resolve this though if you don't think it's worth it.